sale_variant_configurator/models/sale_order.py

```diff
@@ -50,6 +55,27 @@ class SaleOrderLine(models.Model):
         ),
     ]
 
+    @api.depends("product_tmpl_id")
+    def _compute_product_uom(self):
+        lines_with_template = self.filtered(
+            lambda x: x.product_tmpl_id and not x.product_id
+        )
+        for line in lines_with_template:
+            if not line.product_uom or (
+                line.product_tmpl_id.uom_id.id != line.product_uom.id
+            ):
+                line.product_uom = line.product_tmpl_id.uom_id
+        return super(SaleOrderLine, self - lines_with_template)._compute_product_uom()
+
+    @api.depends("product_tmpl_id", "product_id")
+    def _compute_product_uom_category_id(self):
+        for line in self:
+            product = line.product_id or line.product_tmpl_id
+            if product:
+                line.product_uom_category_id = product.uom_id.category_id
+            else:
+                line.product_uom_category_id = line.product_uom_category_id
```

Comment:
Is this still needed in 16.0?

Reply:
Yes, it is a compute no store and it is necessary that all records have a value to avoid the CacheMis error.